assets/js/admin-autoshare-for-twitter.js

```diff
@@ -120,18 +122,38 @@
 	 * Updates the counter
 	 */
 	function updateRemainingField() {
-		var count = $tweetText.val().length;
+		let permalinkLength = 0;
+		if ( $('#sample-permalink').length ) {
+			permalinkLength = $('#sample-permalink').text().length
+		}
+		// +5 because of the space between body and URL and the ellipsis.
+		permalinkLength += 5;
+
+		var count = $tweetText.val().length + permalinkLength;
+		$tweetText.attr('maxlength', limit - permalinkLength);
 
 		$(counterWrap).text(count);
 
 		// Toggle the .over-limit class.
-		if (limit < count) {
+		if (limit <= count) {
+			counterWrap.classList.remove('near-limit');
 			counterWrap.classList.add('over-limit');
-		} else if (counterWrap.classList.contains('over-limit')) {
+			$(counterWrap).text( sprintf( __( '%d - Too Long!', 'autoshare-for-twitter' ), count ) );
+		} else if (240 <= count) {
+			counterWrap.classList.remove('over-limit');
+			counterWrap.classList.add('near-limit');
+			$(counterWrap).text( sprintf( __( '%d - Getting Long!', 'autoshare-for-twitter' ), count ) );
```

Comment:
@iamdharmesh Can document `%d` in `translator comment`. It is helpful in translation plugins.